nltk/parse/stanford.py

```diff
@@ -9,117 +9,177 @@
 
-        model_jar=max(
+        if port is None:
+            try:
+                port = try_port(9000)
+            except socket.error:
+                port = try_port()
+
+            if corenlp_options:
+                corenlp_options.append(str(port))
+            else:
+                corenlp_options = [str(port)]
+        else:
+            try_port(port)
+
+        self.url = 'http://localhost:{}'.format(port)
+
+        model_jar = max(
             find_jar_iter(
-                self._MODEL_JAR_PATTERN, path_to_models_jar,
+                self._MODEL_JAR_PATTERN,
+                path_to_models_jar,
                 env_vars=('STANFORD_MODELS', 'STANFORD_CORENLP'),
-                searchpath=(), url=_stanford_url,
-                verbose=verbose, is_regex=True
+                searchpath=(),
+                url=_stanford_url,
+                verbose=verbose,
+                is_regex=True,
             ),
             key=lambda model_name: re.match(self._MODEL_JAR_PATTERN, model_name)
         )
 
-        #self._classpath = (stanford_jar, model_jar)
-        
-        # Adding logging jar files to classpath 
-        stanford_dir = os.path.split(stanford_jar)[0]
-        self._classpath = tuple([model_jar] + find_jars_within_path(stanford_dir))
-
-        self.model_path = model_path
-        self._encoding = encoding
-        self.corenlp_options = corenlp_options
-        self.java_options = java_options
-
-    def _parse_trees_output(self, output_):
-        res = []
-        cur_lines = []
-        cur_trees = []
-        blank = False
-        for line in output_.splitlines(False):
-            if line == '':
-                if blank:
-                    res.append(iter(cur_trees))
-                    cur_trees = []
-                    blank = False
-                elif self._DOUBLE_SPACED_OUTPUT:
-                    cur_trees.append(self._make_tree('\n'.join(cur_lines)))
-                    cur_lines = []
-                    blank = True
-                else:
-                    res.append(iter([self._make_tree('\n'.join(cur_lines))]))
-                    cur_lines = []
-            else:
-                cur_lines.append(line)
-                blank = False
-        return iter(res)
+        self.other_jars = tuple(find_jars_within_path(os.path.dirname(stanford_jar)))
 
-    def parse_sents(self, sentences, verbose=False):
-        """
-        Use StanfordParser to parse multiple sentences. Takes multiple sentences as a
-        list where each sentence is a list of words.
-        Each sentence will be automatically tagged with this StanfordParser instance's
-        tagger.
-        If whitespaces exists inside a token, then the token will be treated as
-        separate tokens.
+        self.verbose = verbose
+
+        self._classpath = (stanford_jar, model_jar) + self.other_jars
+
+        self.corenlp_options = corenlp_options or []
+        self.java_options = java_options or ['-mx4g']
```

Comment:
If you're only running the POS tagger and a parser (as annotators with significant model sizes), then you can get away with much less RAM for the JVM than this.  -mx1g is likely fine or you could do -mx1500m . The big advantage of this is that the current code would immediately fail for the (admittedly shrinking) group of people using a 32 bit OS, whereas with a lower RAM limit, it'd run fine either way.